### ISSUE TITLE
Feature Request: White-list non-Boom errors / Pass-through GraphQL errors / Override `internal server error`

### ISSUE THREAD

**Issue by KeithGillette** [GiladShoham/graphql-apollo-errors#4]
I have found `graphql-apollo-errors` very useful to simplify error handling in our Apollo-based project. Thanks for publishing this package!

One shortcoming I've run into is the inability to get error messages from Apollo itself (such as `cannot query field` and similar for malformed operations) or error messages from third-party queries and mutations loaded via `graphql-loader`. I certainly understand that you want to prevent exposing general uncaught application exceptions, but providing some mechanism for allowing non-Boom errors to pass-through would be extremely useful.

Perhaps the simplest and most dangerous option would be to just include a `formatErrorGenerator` configuration flag to override the default behavior and allow non-Boom errors to pass-through. If there's an easy way to know whether the error is coming from Apollo itself or from a `graphql-loader` originated operation, then I'd think it would be safe to just pass those through, as well. Otherwise, providing a function hook to white-list errors would be another option.

**Comment by GiladShoham** [comment]
Thanks for the feedback @KeithGillette 
Sorry for the long delay in my reply, I'm currently investing all my time in another very big open source - [bit](https://github.com/teambit/bit) so I haven't been here lately.

I understand your need.
I think we can add transofrmFunction which can be passed as an option, we can add a default one which just wraps error with Boom, but then you can implement your own transformer which run some checks before the transform.
What do you think?

May I ask you to create a PR for this?

Also, I'm planning to share this code as a bit component soon on [https://bitsrc.io](bitsrc.io) which will give you the option to import it into your own project then change and export it with modifications to your own scope.

**Comment by KeithGillette** [comment]
Thanks for the reply, @GiladShoham. I'd be happy to help implement this enhancement, though my own availability is similarly limited by my own project. 

I think I understand the approach you are suggesting at a high level, though I am not familiar enough with the internals of GraphQL-Apollo-Errors nor Apollo server to execute independently. If I understand your suggestion, `formatErrorOptions` would get an additional `transformFunction` property, a user-provided function which I assume would be passed the raw error provided by Apollo and return (throw?) a SevenBoom error. If `transformFunction` was not provided, then the current GraphQL-Apollo-Error implementation would be changed to wrap the Apollo-provided error with a generic SevenBoom error (perhaps `SevenBoom.badImplementation`).

Am I understanding your recommended approach?

**Comment by GiladShoham** [comment]
Thank @KeithGillette 
Yes something like this, expect I think that if the function not provided we should fallback to the existing behavior (for backward compatibility)
I'm talking about changing this line:
https://github.com/GiladShoham/graphql-apollo-errors/blob/master/lib/index.js#L51
to something like:
```js
if (transformFunc && _.isFunction(transformFunc){
  err = transformFunc(err)
} else {
  err = SevenBoom.wrap(err, 500);
}
```

**Comment by gdi2290** [comment]
if you want this feature we published a fork `@tipe/graphql-apollo-errors` until it's in

**Comment by GiladShoham** [comment]
@KeithGillette A new version has been published (v2.0.3) containing your PR code.
Please check and let me know if I can close this issue.
@gdi2290  FYI.

Feel free to open more issues / PRs.
Thanks.

**Comment by KeithGillette** [comment]
Works for me! Thanks for accepting my contribution, @GiladShoham!
